Homework3

Kiba23 opened the pull request:
First Pull Request for Homework3. Not reviewed by anybody.

Kiba23 committed:
Init commit

Just started the project. Initialized files and deleted old folders of recent homeworks.

Kiba23 committed:
First breakpoint.

Added some logic and input checker.

Kiba23 committed:
Second breakpoint with completed Equals method.

Implemented all the logic except CompareTo method. Method Equals now works in all possible cases.

Kiba23 committed:
Delete Homework3/.vs/Homework3 directory

Kiba23 committed:
Delete Homework3/PianoKeyboard/obj directory

Kiba23 committed:
Delete Homework3/PianoKeyboard/bin/Debug/net5.0 directory

Kiba23 committed:
Finished the task.

Implemented the CompareTo method. Added some comments.

Kiba23 committed:
Merge branch 'homework3' of https://github.com/Kiba23/Coherent-Hometasks into homework3

Kiba23 committed:
Completed Second task. Init commit.

First commit on second task and already done. Implemented all the things. Added comments and tests in Program.cs file.

radislavkireev commented:
FYI - there were no need to delete HW1, HW2, you had to merge approved PRs. 

No actions required, just keep in mind pls.

radislavkireev commented on Homework3/StackDataType/Stack.cs:
```diff
@@ -0,0 +1,61 @@
+        public const int MAX_STACK_SIZE = 10;
+
+        public int LastElementIndex { get; private set; } = -1; // -1 because of first value in array should be at index 0.
+        T[] Data = new T[MAX_STACK_SIZE];
```
make Data private

radislavkireev commented on Homework3/StackDataType/Stack.cs:
```diff
@@ -0,0 +1,61 @@
+        }
+        public bool IsEmpty()
+        {
+            return LastElementIndex < 0 ? true : false;
```
simplify to `
  return LastElementIndex < 0 `

radislavkireev commented on Homework3/StackDataType/Stack.cs:
```diff
@@ -0,0 +1,61 @@
+        {
+            return LastElementIndex < 0 ? true : false;
+        }
+        public Stack<T> DeepCopy()
```
Drop this method, it is not needed

radislavkireev commented on Homework3/StackDataType/Interfaces/IStack.cs:
```diff
@@ -0,0 +1,13 @@
+{
+    public interface IStack<T>
+    {
+        int LastElementIndex { get; }
```
Drop this property, it should not be part of interface per requirements

radislavkireev commented on Homework3/StackDataType/Interfaces/IStack.cs:
```diff
@@ -0,0 +1,13 @@
+        void Push(T value);
+        T Pop();
+        bool IsEmpty();
+        Stack<T> DeepCopy();
```
Drop this property, it should not be part of interface per requirements

radislavkireev commented on Homework3/StackDataType/IStackExtension.cs:
```diff
@@ -0,0 +1,33 @@
+            return resultStack;
+        }
+        // This is in case you want to leave stack empty after this method, because Pop method deletes the values from original stack.
+        public static IStack<T> ReverseAndClearStack<T>(this IStack<T> stack) 
```
Keep this method only, and name it Reverse

radislavkireev commented on Homework3/StackDataType/IStackExtension.cs:
```diff
@@ -0,0 +1,33 @@
+        {
+            IStack<T> newStack = new Stack<T>();
+
+            for (int i = stack.LastElementIndex; i >= 0; i--)
```
LastElementIndex should not exist, use IsEmpty for checking end of stack. .

Kiba23 committed:
Fixed HW3 after Radislav's suggestions.

Droped some unnecessary methods and properties (can see in IStack<T> interface). Droped additional Reverse method. Fixed property in Stack class.
